designer/server/src/main/scala/pl/touk/nussknacker/ui/api/BaseHttpService.scala

```diff
@@ -0,0 +1,60 @@
+package pl.touk.nussknacker.ui.api
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.ui.process.ProcessCategoryService
+import pl.touk.nussknacker.ui.security.api._
+import sttp.tapir.AnyEndpoint
+import sttp.tapir.server.ServerEndpoint
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.{ExecutionContext, Future}
+
+abstract class BaseHttpService(config: Config,
+                               processCategoryService: ProcessCategoryService,
+                               authenticator: AuthenticationResources)
+                              (implicit executionContext: ExecutionContext) {
+
+  private val allServerEndpoints = new AtomicReference(List.empty[ServerEndpoint[Any, Future]])
+
+  protected def expose(serverEndpoint: ServerEndpoint[Any, Future]): Unit = {
+    allServerEndpoints
+      .accumulateAndGet(
+        List(serverEndpoint),
+        (l1, l2) => l1 ::: l2
+      )
+  }
+
+  protected def expose(when: => Boolean)(serverEndpoint: ServerEndpoint[Any, Future]): Unit = {
+    if(when) expose(serverEndpoint)
+  }
+
+  def allEndpointDefinitions: List[AnyEndpoint] = serverEndpoints.map(_.endpoint)
+
+  def serverEndpoints: List[ServerEndpoint[Any, Future]] = allServerEndpoints.get()
+
+  protected def authorizeAdmin[ERROR](credentials: AuthCredentials): Future[Either[Either[ERROR, SecurityError], LoggedUser]] = {
+    authorize[ERROR](credentials)
+      .map {
+        case right@Right(AdminUser(_, _)) => right
+        case Right(_: CommonUser) => Left(Right(SecurityError.AuthorizationError))
+        case error@Left(_) => error
+      }
+  }
+
+  protected def authorize[ERROR](credentials: AuthCredentials): Future[Either[Either[ERROR, SecurityError], LoggedUser]] = {
+    authenticator
+      .authenticate(credentials)
+      .map {
+        case Some(user) if user.roles.nonEmpty =>
```

Comment:
The logic is the same as it was before. Here is the authorization of the NU Common User. It's a user with a non-empty list of roles. Most of the authorization logic is placed in the `LoggedUser` class. I decided to not touch it. 

But I've changed the protected `authorizeXXX` methods' names to be more meaningful.

Reply:
I understand that logic is the same, but we have to rename this method. "authorizing" is the less important, more important things are: which resource we authorise access to and how we authorise.

Reply:
The method of authorization is unknown for this class because we use the implementation in LoggedUser object now. So I don't think we can add the "how" part to the name (without refactoring of LoggedUser class of course). 

WDYT about my proposition? (see code)